In [1]:
%config Completer.use_jedi = False
import warnings
warnings.filterwarnings(action="ignore")

In [2]:
import requests
import folium
#!pip install pandas # 파이썬 데이터 처리 라이브러리 pandas를 설치하고 import한다.
import pandas as pd
#파이썬의 딕셔너리 타입의 데이터를 pandas 데이터 프레임으로 변환하기위해 import 한다.
from pandas.io.json import json_normalize

전국 또는 특정 지역의 스타벅스 매장 위장 위치 정볼르 가져와 지도 위에 시각화하기

1. requests 모듈로 스타벅스 매장 위치 정보를 가져와서 딕셔너리 타입으로 변경한다.  
2. json_normalize() 함수를 사용해서 json 타입의 데이터가 변환된 딕셔너리를 데이터프레임으로 변환한다.  
3. folium 모듈을 사용해서 지도를 만들고 지도 위에 스타벅스 매장 위치를 marker로 표시한다.  

In [3]:
#서울특별시 종로구 스타벅스 전지점 위치 크롤링
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "",
    "p_gugun_cd" : "0118", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

store = starbucks["list"]
for i in  range(len(store)):
    print("{:4d}. {}: {}({}, {})".format( (i + 1), store[i]["s_name"], store[i]["doro_address"], store[i]["lat"], store[i]["lot"]))

   1. 서대문역: 서울특별시 종로구 통일로 134 (평동)(37.56670, 126.96629)
   2. 서울대치과병원: 서울특별시 종로구 대학로 101 서울대학교 치과병원 신관1층(37.5774740926489, 126.999891434786)
   3. 동숭길입구: 서울특별시 종로구 대학로 146 (혜화동)(37.584236, 127.002014)
   4. 강북삼성병원1층: 서울특별시 종로구 새문안로 29 (평동)(37.568546, 126.96756)
   5. 종로플레이스: 서울특별시 종로구 창경궁로 120 (인의동)(37.5731377743929, 126.998354975801)
   6. 정동: 서울특별시 종로구 새문안로 42 (신문로2가)(37.568578, 126.970067)
   7. 광화문교보문고: 서울특별시 종로구 종로 1 (종로1가)(37.571017, 126.977937)
   8. 종로관수: 서울특별시 종로구 삼일대로20길 13 (관수동)(37.569385, 126.988665)
   9. 경복궁사거리: 서울특별시 종로구 율곡로 6, B동 101호(중학동, 트윈트리타워)(37.575578, 126.980581)
  10. 대학로: 서울특별시 종로구 대학로 57, 홍익대학교 대학로캠퍼스 內 (연건동)(37.5764916392943, 127.001470988378)
  11. 종로3가: 서울특별시 종로구 종로 113-1, 1-4층 (종로3가)(37.5705527, 126.9902055)
  12. 마로니에공원: 서울특별시 종로구 대학로 116, 1-2층 (동숭동)(37.58154296, 127.0023412)
  13. 이마빌딩: 서울특별시 종로구 종로1길 42 (수송동)(37.5738637, 126.97897149999994)
  14. 경복궁역: 서울특별시 종로구 자하문로4길 6, 1층 (통의동)(37.5771508, 126.97274530000004)
  15. 종로관철: 서울특별시 종로구 종로12길 21, 2층 (관철동)(

In [4]:
#pandas.io.json 모듈의 json_normalize() 함수로 json 타입의 데이터가 변환된 딕셔너리를
#pandas 데이터 프레임으로 변환한다.
#json_normalize(딕셔너리, "데이터프레임으로 변환할 데이터가 할당된 딕셔너리의 key")
starbucks_df = json_normalize(starbucks, "list")
print(type(starbucks_df)) #<class 'pandas.core.frame.DataFrame'>
print(len(starbucks_df))
print(starbucks_df.columns)
starbucks_df

<class 'pandas.core.frame.DataFrame'>
39
Index(['seq', 'sido_cd', 'sido_nm', 'gugun_cd', 'gugun_nm', 'code_order',
       'view_yn', 'store_num', 'sido', 'gugun',
       ...
       'p01', 't05', 't30', 't36', 't27', 't29', 't43', 't48', 'z9999', 'p02'],
      dtype='object', length=134)


seq sido_cd sido_nm gugun_cd gugun_nm code_order view_yn store_num  sido  \
0     0    None    None     None     None       None    None      None  None   
1     0    None    None     None     None       None    None      None  None   
2     0    None    None     None     None       None    None      None  None   
3     0    None    None     None     None       None    None      None  None   
4     0    None    None     None     None       None    None      None  None   
5     0    None    None     None     None       None    None      None  None   
6     0    None    None     None     None       None    None      None  None   
7     0    None    None     None     None       None    None      None  None   
8     0    None    None     None     None       None    None      None  None   
9     0    None    None     None     None       None    None      None  None   
10    0    None    None     None     None       None    None      None  None   
11    0    None    None     None     None       None    None      None  None   
12    0    None    None     None     None       None    None      None  None   
13    0    None    None     None     None       None    None      None  None   
14    0    None    None     None     None       None    None      None  None   
15    0    None    None     None     None       None    None      None  None   
16    0    None    None     None     None       None    None      None  None   
17    0    None    None     None     None       None    None      None  None   
18    0    None    None     None     None       None    None      None  None   
19    0    None    None     None     None       None    None      None  None   
20    0    None    None     None     None       None    None      None  None   
21    0    None    None     None     None       None    None      None  None   
22    0    None    None     None     None       None    None      None  None   
23    0    None    None     None     None       None    None      None  None   
24    0    None    None     None     None       None    None      None  None   
25    0    None    None     None     None       None    None      None  None   
26    0    None    None     None     None       None    None      None  None   
27    0    None    None     None     None       None    None      None  None   
28    0    None    None     None     None       None    None      None  None   
29    0    None    None     None     None       None    None      None  None   
30    0    None    None     None     None       None    None      None  None   
31    0    None    None     None     None       None    None      None  None   
32    0    None    None     None     None       None    None      None  None   
33    0    None    None     None     None       None    None      None  None   
34    0    None    None     None     None       None    None      None  None   
35    0    None    None     None     None       None    None      None  None   
36    0    None    None     None     None       None    None      None  None   
37    0    None    None     None     None       None    None      None  None   
38    0    None    None     None     None       None    None      None  None   

   gugun  ... p01 t05  t30 t36 t27 t29 t43 t48 z9999 p02  
0   None  ...   0   0    0   0   0   0   0   0     0   0  
1   None  ...   0   0    0   0   0   0   0   0     0   0  
2   None  ...   0   0    0   0   0   0   0   0     0   0  
3   None  ...   0   0    0   0   0   0   0   0     0   0  
4   None  ...   0   0    0   0   0   0   0   0     0   0  
5   None  ...   0   0    0   0   0   0   0   0     0   0  
6   None  ...   0   0    0   0   0   0   0   0     0   0  
7   None  ...   0   0    0   0   0   0   0   0     0   0  
8   None  ...   0   0    0   0   0   0   0   0     0   0  
9   None  ...   0   0    0   0   0   0   0   0     0   0  
10  None  ...   0   0    0   0   0   0   0   0     0   0  
11  None  ...   0   0    0   0   0   0   0   0     0   0  
12  None  ...   0   0    0   0   0  

#딕셔너리를 데이터프레임으로 변환하니 서울특별시 종로구에 있는 스타벅스 지점은 40개고,  
#각 지점당 134개의 컬럼으로 표시하고 있다.  
#작업에 필요한 컬럼 몇가지를 선택해서 지도에 마커를 표시할 때 사용할 데이터 프레임을 만든다.  
#s_name -> 지점이름  
#sido_code -> 시도코드  
#sido_name -> 시도 이름  
#gugun_code -> 구군 코드  
#gugun_name -> 구군 이름  
#doro_address -> 도로명 주소  
#lat -> 위도  
#lot -> 경도  

In [5]:
#pandas 데이터 프레임에서 특정 컬럼의 데이터를 얻어올 때 얻어올 컬럼이 1개라면 컬럼 이름만 적어도 되지만,
#얻어올 칼럼이 2개 이상일 경우 얻어올 컬럼들을 반드시 리스트 형태로 만들어서 얻어와야 한다.
starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]
del starbucks_df
starbucks_df_map

s_name sido_code sido_name gugun_code gugun_name  \
0       서대문역        01        서울       0118        종로구   
1    서울대치과병원        01        서울       0118        종로구   
2      동숭길입구        01        서울       0118        종로구   
3   강북삼성병원1층        01        서울       0118        종로구   
4     종로플레이스        01        서울       0118        종로구   
5         정동        01        서울       0118        종로구   
6    광화문교보문고        01        서울       0118        종로구   
7       종로관수        01        서울       0118        종로구   
8     경복궁사거리        01        서울       0118        종로구   
9        대학로        01        서울       0118        종로구   
10      종로3가        01        서울       0118        종로구   
11    마로니에공원        01        서울       0118        종로구   
12      이마빌딩        01        서울       0118        종로구   
13      경복궁역        01        서울       0118        종로구   
14      종로관철        01        서울       0118        종로구   
15     동숭로아트        01        서울       0118        종로구   
16        종각        01        서울       0118        종로구   
17     주얼리시티        01        서울       0118        종로구   
18     스페이스본        01        서울       0118        종로구   
19      성대입구        01        서울       0118        종로구   
20       광화문        01        서울       0118        종로구   
21        인사        01        서울       0118        종로구   
22        적선        01        서울       0118        종로구   
23    광화문D타워        01        서울       0118        종로구   
24       삼청동        01        서울       0118        종로구   
25        안국        01        서울       0118        종로구   
26      새문안로        01        서울       0118        종로구   
27      종로수송        01        서울       0118        종로구   
28       북촌로        01        서울       0118        종로구   
29      종로구청        01        서울       0118        종로구   
30   정부서울청사R        01        서울       0118        종로구   
31      독립문역        01        서울       0118        종로구   
32       종로R        01        서울       0118        종로구   
33    광화문우체국        01        서울       0118        종로구   
34     종로평창동        01        서울       0118        종로구   
35    강북삼성병원        01        서울       0118        종로구   
36      경희궁로        01        서울       0118        종로구   
37       혜화역        01        서울       0118        종로구   
38      연합뉴스        01        서울       0118        종로구   

                                        doro_address                 lat  \
0                             서울특별시 종로구 통일로 134 (평동)            37.56670   
1                  서울특별시 종로구 대학로 101 서울대학교 치과병원 신관1층    37.5774740926489   
2                            서울특별시 종로구 대학로 146 (혜화동)           37.584236   
3                             서울특별시 종로구 새문안로 29 (평동)           37.568546   
4                           서울특별시 종로구 창경궁로 120 (인의동)    37.5731377743929   
5                          서울특별시 종로구 새문안로 42 (신문로2가)           37.568578   
6                              서울특별시 종로구 종로 1 (종로1가)           37.571017   
7                         서울특별시 종로구 삼일대로20길 13 (관수동)           37.569385   
8              서울특별시 종로구 율곡로 6, B동 101호(중학동, 트윈트리타워)           37.575578   
9             서울특별시 종로구 대학로 57, 홍익대학교 대학로캠퍼스 內 (연건동)    37.5764916392943   
10                   서울특별시 종로구 종로 113-1, 1-4층 (종로3가)          37.5705527   
11                     서울특별시 종로구 대학로 116, 1-2층 (동숭동)         37.58154296   
12                           서울특별시 종로구 종로1길 42 (수송동)          37.5738637   
13                      서울특별시 종로구 자하문로4길 6, 1층 (통의동)          37.5771508   
14                      서울특별시 종로구 종로12길 21, 2층 (관철동)           37.569058   
15                           서울특별시 종로구 동숭길 110 (동숭동)           37.582969   
16                            서울특별시 종로구 종로 64 (종로2가)          37.5699183   
17                    서울특별시 종로구 종로 183 (인의동) 효성주얼리시티    37.5714829206881   
18               서울특별시 종로구 사직로8길 4 (사직동) 광화문 풍림스페이스본   37.57488602790947   
19                         서울특별시 종로구 창경궁로 235 (명륜3가)           37.582892   
20                          서울특별시 종로구 세종대로 167 (세종로)          37.5712293   
21           

In [6]:
#dtypes 속성으로 데이터 프레임을 구성하는 컬럼들의 자료형을 확인할 수 있다.
#pandas는 문자열 데이터 타입을 object라 부른다.
starbucks_df_map.dtypes

s_name          object
sido_code       object
sido_name       object
gugun_code      object
gugun_name      object
doro_address    object
lat             object
lot             object
dtype: object

In [7]:
#info() 메소드를 사용해도 데이터프레임을 구성하는 컬럼들의 자료형을 확인할 수 있다.
starbucks_df_map.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39 entries, 0 to 38
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   s_name        39 non-null     object
 1   sido_code     39 non-null     object
 2   sido_name     39 non-null     object
 3   gugun_code    39 non-null     object
 4   gugun_name    39 non-null     object
 5   doro_address  39 non-null     object
 6   lat           39 non-null     object
 7   lot           39 non-null     object
dtypes: object(8)
memory usage: 2.6+ KB


In [8]:
#astype() 메소드로 위도, 경도 데이터 타입을 object에서 float로 변환한다.
starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)
starbucks_df_map.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39 entries, 0 to 38
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   s_name        39 non-null     object 
 1   sido_code     39 non-null     object 
 2   sido_name     39 non-null     object 
 3   gugun_code    39 non-null     object 
 4   gugun_name    39 non-null     object 
 5   doro_address  39 non-null     object 
 6   lat           39 non-null     float64
 7   lot           39 non-null     float64
dtypes: float64(2), object(6)
memory usage: 2.6+ KB


In [9]:
#종각점
starbucks_df_map[starbucks_df_map["s_name"] == "종각"]

s_name sido_code sido_name gugun_code gugun_name            doro_address  \
16     종각        01        서울       0118        종로구  서울특별시 종로구 종로 64 (종로2가)   

          lat         lot  
16  37.569918  126.984528

In [10]:
#종로관철점
starbucks_df_map[starbucks_df_map["s_name"] == "종로관철"]

s_name sido_code sido_name gugun_code gugun_name  \
14   종로관철        01        서울       0118        종로구   

                    doro_address        lat         lot  
14  서울특별시 종로구 종로12길 21, 2층 (관철동)  37.569058  126.986013

In [11]:
#종로구청점
starbucks_df_map[starbucks_df_map["s_name"] == "종로구청"]

s_name sido_code sido_name gugun_code gugun_name            doro_address  \
29   종로구청        01        서울       0118        종로구  서울특별시 종로구 삼봉로 71 (수송동)   

          lat         lot  
29  37.572662  126.981138

In [12]:
foliumMap = folium.Map(location=[37.569147, 126.984695], zoom_start=13)
popup = folium.Popup("스타벅스 종각", max_width=300)
folium.Marker(location=[37.5699183, 126.98452810000003], popup=popup).add_to(foliumMap)
popup = folium.Popup("스타벅스 종로관철", max_width=300)
folium.Marker(location=[37.569058, 126.986], popup=popup).add_to(foliumMap)
popup = folium.Popup("스타벅스 종로구청", max_width=300)
folium.Marker(location=[37.572662, 126.981], popup=popup).add_to(foliumMap)
foliumMap.save("./output/starbucksMap1.htm")
foliumMap

In [13]:
#서울특별시 종로구에 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
#지도를 작성할 때 지도의 중심을 설정할 위치가 애매할 경우, 지도상에 표시할 모든 스타벅스 지점의
#위도와 경도의 평균을 중심으로 하는 지도를 만들어 사용하면 된다.
#데이터프레임에 저장된 데이터 평균은 mean() 메소드를 사용해서 계산한다.
foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=13)

#iterrows() 메소드는 데이터 프레임에 저장된 데이터의 인덱스와 데이터를 튜플 형태로 리턴한다.
for index, data in starbucks_df_map.iterrows():
    #print(index, data)
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)


foliumMap.save("./output/starbucksMap2.htm")
foliumMap

In [14]:
#서울특별시 강남구에 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "",
    "p_gugun_cd" : "0101", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

starbucks_df = json_normalize(starbucks, "list")

starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]

starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)

foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=13)

for index, data in starbucks_df_map.iterrows():
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)

foliumMap.save("./output/starbucksMap3.htm")
foliumMap

In [15]:
#서울특별시에 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "01",
    "p_gugun_cd" : "", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

starbucks_df = json_normalize(starbucks, "list")

starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]

starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)

foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=11)

for index, data in starbucks_df_map.iterrows():
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)

foliumMap.save("./output/starbucksMap4.htm")
foliumMap

In [16]:
#경기도에 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "08",
    "p_gugun_cd" : "", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

starbucks_df = json_normalize(starbucks, "list")

starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]

starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)

foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=10)

for index, data in starbucks_df_map.iterrows():
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)

foliumMap.save("./output/starbucksMap5.htm")
foliumMap

In [17]:
#제주도에 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "16",
    "p_gugun_cd" : "", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

starbucks_df = json_normalize(starbucks, "list")

starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]

starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)

foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=11)

for index, data in starbucks_df_map.iterrows():
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)

foliumMap.save("./output/starbucksMap6.htm")
foliumMap

In [19]:
#전국 위치한 스타벅스 전 지점의 위치를 지도위에 표시한다.
targetSite = "https://www.starbucks.co.kr:7643/store/getStore.do?r=M05ESFPDLV"
request = requests.post(targetSite, data={
   "ins_lat": 37.5691335,
    "ins_lng" :	126.9847584, 
    "p_sido_cd" : "",
    "p_gugun_cd" : "", 
    "in_biz_cd" : "",
    "iend" : 1000,
    "set_date" : ""
})
starbucks = request.json()

starbucks_df = json_normalize(starbucks, "list")

starbucks_df_map = starbucks_df[
    [
    "s_name",
    "sido_code",
    "sido_name",
    "gugun_code",
    "gugun_name",
    "doro_address",
    "lat",
    "lot"
    ]
]

starbucks_df_map["lat"] = starbucks_df_map["lat"].astype(float)
starbucks_df_map["lot"] = starbucks_df_map["lot"].astype(float)

foliumMap = folium.Map(location=[starbucks_df_map["lat"].mean(), starbucks_df_map["lot"].mean()], zoom_start=8)

for index, data in starbucks_df_map.iterrows():
    popup = folium.Popup("{}. {} - {}".format((index + 1),data["s_name"],data["doro_address"]), max_width=300)
    folium.Marker(location=[data["lat"], data["lot"]], popup=popup).add_to(foliumMap)

foliumMap.save("./output/starbucksMap7.htm")
foliumMap